src/components/ContextBar.test.js

```diff
@@ -0,0 +1,625 @@
+      persistentDisks: [runtimeDisk]
+    }
+
+    // Act
+    const { getByText, getByLabelText } = render(h(ContextBar, jupyterContextBarProps))
+    fireEvent.click(getByLabelText(new RegExp(/Jupyter Environment/i)))
+
+    // Assert
+    getByText('Cloud Environment Details')
+    getByText(tools.Jupyter.label)
+  })
+  it('clicking Galaxy opens CloudEnvironmentModal with Galaxy as filter for tool.', () => {
+    // Arrange
+    const galaxyContextBarProps = {
+      ...contextBarProps,
+      apps: [galaxyRunning],
+      appDataDisks: [galaxyDisk]
+    }
+
+    // Act
+    const { getByLabelText, getByText } = render(h(ContextBar, galaxyContextBarProps))
+    fireEvent.click(getByLabelText(new RegExp(/Galaxy Environment/i)))
+
+    // Assert
+    getByText('Cloud Environment Details')
+    getByText(tools.Galaxy.label)
+  })
+  it('clicking RStudio opens CloudEnvironmentModal with RStudio as filter for tool.', () => {
+    // Act
+    const rstudioContextBarProps = {
+      ...contextBarProps,
+      runtimes: [rstudioRuntime],
+      apps: [galaxyRunning],
+      appDataDisks: [galaxyDisk],
+      persistentDisks: [runtimeDisk]
+    }
+
+    // Act
+    const { getByText, getByLabelText } = render(h(ContextBar, rstudioContextBarProps))
+    fireEvent.click(getByLabelText(new RegExp(/RStudio Environment/i)))
+
+    // Assert
+    getByText('Cloud Environment Details')
+    getByText(tools.RStudio.label)
+  })
+
+  it('clicking Terminal will attempt to start currently stopped runtime', () => {
+    // Arrange
+    global.window = Object.create(window)
+    const url = 'http://dummy.com'
+    Object.defineProperty(window, 'location', {
+      value: {
+        href: url
+      },
+      writable: true,
+      hash: '/'
+    })
+    const jupyterContextBarProps = {
+      ...contextBarProps,
+      runtimes: [{
+        ...jupyter,
+        status: 'Stopped'
+      }],
+      persistentDisks: [runtimeDisk]
+    }
+
+    // Act
+    const { getByLabelText } = render(h(ContextBar, jupyterContextBarProps))
+    fireEvent.click(getByLabelText('Terminal button'))
+
+    // Assert
+    expect(Ajax().Runtimes.runtime).toBeCalledWith(jupyter.googleProject, jupyter.runtimeName)
+    expect(mockRuntimesStartFn).toBeCalledTimes(1)
+  })
+
+  it('clicking Terminal will not attempt to start an already running Jupyter notebook', () => {
+    // Arrange
+    global.window = Object.create(window)
+    const url = 'http://dummy.com'
+    Object.defineProperty(window, 'location', {
+      value: {
+        href: url
+      },
+      writable: true,
+      hash: '/'
+    })
+    const jupyterContextBarProps = {
+      ...contextBarProps,
+      runtimes: [
+        jupyter
+      ],
+      persistentDisks: [runtimeDisk]
+    }
+
+    // Act
+    const { getByLabelText } = render(h(ContextBar, jupyterContextBarProps))
+    fireEvent.click(getByLabelText('Terminal button'))
+
+    // Assert
+    expect(mockRuntimesStartFn).toBeCalledTimes(0)
```

Comment:
Either way is fine, but I think `expect(mockRuntimesStartFn).not.toHaveBeenCalled()` reads a little more clearly for things like this.